subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -298,7 +308,6 @@ private DefaultGroovyJavaJointCompileSpec createSpec() {
         spec.setGroovyCompileOptions(groovyCompileOptions);
         if (enableIncrementalCompilation(spec)) {
```

Comment:
Can we pass in the source roots to `enableIncrementalCompilation`? Same for `annotationProcessingConfigured` I guess.

I find it currently strange that we use `GroovyCompile.option.incremental` and the other two things from the spec while we are still in the process of modifying the spec.

Reply:
Better now?

Reply:
Yeah. I wonder if we can avoid the duplicated logic for checking if there is some annotation processing configured.

Reply:
I really don't want to extract another method, so I'll just pass `boolean annotationProcessorConfigured` to the method.